### PR TITLE
Fix node port tests windows nodeSelector metadata

### PR DESCRIPTION
**Pull request by jayunit100**
Fix to #97807  , similar to #97805

### CODE REVIEW

File: test/e2e/windows/service.go [hunk 1]
```diff
@@ -61,11 +61,14 @@ var _ = SIGDescribe("Services", func() {
 
 		//using hybrid_network methods
 		ginkgo.By("creating Windows testing Pod")
-		windowsPod := createTestPod(f, windowsBusyBoximage, windowsOS)
-		windowsPod = f.PodClient().CreateSync(windowsPod)
+		testPod := createTestPod(f, windowsBusyBoximage, windowsOS)
```
Comment (jsturtevant) [inline test/e2e/windows/service.go:64]: CreateTestPod sets the OS on the node selector: 

https://github.com/kubernetes/kubernetes/blob/3552bb5d8a893a8168cfa23001342aaebcd311d1/test/e2e/windows/hybrid_network.go#L103-L124
Do we know why it's not being respected?

File: test/e2e/windows/dns.go [hunk 2]
```diff
@@ -50,6 +50,9 @@ var _ = SIGDescribe("DNS", func() {
 			Nameservers: []string{testInjectedIP},
 			Searches:    []string{testSearchPath},
 		}
+		testUtilsPod.Spec.NodeSelector = map[string]string{
```
Comment (jsturtevant) [inline test/e2e/windows/dns.go:53]: this was added in https://github.com/kubernetes/kubernetes/pull/97804

File: test/e2e/windows/service.go (hunk 1 above)
Comment (jayunit100) [reply test/e2e/windows/service.go:64]: good catch ! ok ya we discussed that yesterday :) ... ok let me add a ginkgo.By to this PR, then do more testing, then revisit this in a few

Comment (jayunit100) [reply test/e2e/windows/service.go:64]: wait ! its in `jig.Run` changing this nowwww

File: test/e2e/windows/dns.go (hunk 2 above)
Comment (jayunit100) [reply test/e2e/windows/dns.go]: yup rebased just now

File: test/e2e/windows/service.go (hunk 1 above)
Comment (jayunit100) [reply test/e2e/windows/service.go:70]: ive backed this out into a ginkgo.By, rather then a redundant os.windows step

File: test/e2e/windows/service.go [hunk 3]
```diff
@@ -56,16 +56,26 @@ var _ = SIGDescribe("Services", func() {
 		nodePort := int(svc.Spec.Ports[0].NodePort)
 
 		ginkgo.By("creating Pod to be part of service " + serviceName)
-		_, err = jig.Run(nil)
+		// tweak the Jig to use windows...
+		windowsTweak := func(rc *v1.ReplicationController) {
```
Comment (jsturtevant) [inline test/e2e/windows/service.go:60]: nit: can we call this windowsNodeSelector or something like that?

Comment (jayunit100) [reply test/e2e/windows/service.go:60]: sure ? whatever is fine, did you wanna use that fancy github `insert a suggestion` button and ill just merge your commit as part of this?  or i can push one ?

Comment (jayunit100) [reply test/e2e/windows/service.go:60]: ( i actually like windowsTweak bc, literally, the Jig names the input variable `tweak`) so would have a +.00001 preference to keep it named as is just for readability

Comment (jayunit100) [reply test/e2e/windows/service.go]: pushed a rename , `windowsNodeSelectorTweak` which probably satisfies all of our needs :)

Comment (jsturtevant) [reply test/e2e/windows/service.go]: cool TIL the input variable is called tweak :-)

**Review by michmike — APPROVED** [review]
should we also be checking for a toleration added?
/lgtm
/approve

**Pull request merged**